File: app/src/main/java/com/berke/sozluk/DatabaseAccess.java
package com.berke.sozluk;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;


public class DatabaseAccess
{
    private SQLiteOpenHelper openHelper;
    private SQLiteDatabase database;
    private static DatabaseAccess instance;
    private String[] guessedWords;
    private List<Thread> threadList = new ArrayList<>();

    private DatabaseAccess(Context context)
    {
        this.openHelper = new DatabaseOpenHelper(context);
    }

    public static DatabaseAccess getInstance(Context context) {
        if( instance == null) {
            instance = new DatabaseAccess(context);
        }
        return instance;
    }

    public void open()
    {
        this.database = openHelper.getWritableDatabase();
    }

    public void close()
    {
        if(database != null)
        {
            this.database.close();
        }
    }

    public String getDefinition(String word)
    {
        Cursor cursor = database.rawQuery("SELECT meaning FROM words WHERE TRIM(word) = '"+ word + "' COLLATE NOCASE",null);
        cursor.moveToFirst();

        if(cursor.getCount() > 0)
        {
            String definition = cursor.getString(0);
            cursor.close();
            return definition;
        }
        return null;
    }
    public String[] getSuggestions(String word)
    {

        String[] words= new String[5];
        String dbWord = "" + word +"%";
        Cursor cursor = database.rawQuery("SELECT word FROM words WHERE TRIM(word) LIKE ? LIMIT 5",new String[]{dbWord});
        cursor.moveToFirst();
        for(int i =0; i<words.length && i < cursor.getCount(); i++)
        {
            words[i] = cursor.getString(0);
            cursor.moveToNext();
        }
        cursor.close();
        return words;
    }

}

File: app/src/main/java/com/berke/sozluk/DatabaseAccess.kt
package com.berke.sozluk

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DatabaseAccess
private constructor(context: Context) {
    private val openHelper: SQLiteOpenHelper = DatabaseOpenHelper(context)
    private var database: SQLiteDatabase? = null

    fun open() {
        this.database = openHelper.writableDatabase
    }

    fun close() {
        if (database != null) {
            database!!.close()
        }
    }

    fun getDefinition(word: String): String? {
        val cursor = database!!.rawQuery(
            "SELECT meaning FROM words WHERE TRIM(word) = '$word' COLLATE NOCASE",
            null
        )
        cursor.moveToFirst()

        if (cursor.count > 0) {
            val definition = cursor.getString(0)
            cursor.close()
            return definition
        }
        return null
    }

    fun getSuggestions(word: String): Array<String?> {
        val words = arrayOfNulls<String>(5)
        val dbWord = "$word%"
        val cursor = database!!.rawQuery(
            "SELECT word FROM words WHERE TRIM(word) LIKE ? LIMIT 5",
            arrayOf(dbWord)
        )
        cursor.moveToFirst()
        var i = 0
        while (i < words.size && i < cursor.count) {
            words[i] = cursor.getString(0)
            cursor.moveToNext()
            i++
        }
        cursor.close()
        return words
    }

    companion object {
        private var instance: DatabaseAccess? = null

        fun getInstance(context: Context): DatabaseAccess? {
            if (instance == null) {
                instance = DatabaseAccess(context)
            }
            return instance
        }
    }
}